File: app/channels/chat_channel.rb
# frozen_string_literal: true

class ChatChannel < ApplicationCable::Channel
	def subscribed
		stream_from "chat:#{params[:chat_id]}"
		puts "STREAMING FROM chat:#{params[:chat_id]}"
	end

	# def receive(data)
	# 	ActionCable.server.broadcast('test', 'ActionCable is connected')
	# end

	def unsubscribed
		# Any cleanup needed when channel is unsubscribed
	end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer";

export default function connectToChatChannel(chatId) {
  consumer.subscriptions.create({
    channel: 'ChatChannel',
    chat_id: chatId
  }, {
    connected() {
      console.log(`Connecting to chat Channel for chat ${chatId}`)
      // Called when the subscription is ready for use on the server
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received(data) {
      console.warn(data);
      if (data.cableReady) CableReady.perform(data.operations);
    }
  });
}
